docs/diagrams/EditExecuteRefrence.puml

```diff
@@ -0,0 +1,57 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+!include Styles.puml
+
+box model COLOR_MODEL_SEGMENT
+participant ":Stock" as Instrument COLOR_MODEL
+end box
+
+activate Instrument
+group sd check if parameter exist in HashMap and edit
+
+Instrument ->Instrument: editGeneralParameter({"name, new name"})
+activate Instrument
+note left: This method contains editName, editCurrentPrice,\neditSentiment and editDoneStatus,\nwhich are similar methods.
+Instrument -> Instrument: editName({"name, new name"})
+note left: This method check is the key "name" exist in HashMap\nand if it exist, set name parameter to new name
+activate Instrument
+
+alt HashMap does not contain the key "name"
+else HashMap contains the key "name"
```

Comment:
Forget the alt here, it doesnt seem to add much value.

Reply:
Rather create the self invocation for all the attributes like current price and sentiment under editGeneralParameters